### PR TITLE
fix: improve log display performance

### LINKED ISSUE #360: Add pagination on log per task (gantt & logs tabs)
Some task can have a lot of log, and will freeze the browser. 
We need to add some pagination and some filter (text at min)

### LINKED ISSUE #1252: Download & Copy Logs buttons are not working
### Expected Behavior

When clicking on the button I would expect to get files downloaded locally (download button) or logs copied in my clipboard (copy button)
image

### Actual Behaviour

_No response_

### Steps To Reproduce

_No response_

### Environment Information

- Kestra Version: 0.9.0
- Operating System (OS / Docker / Kubernetes): Docker
- Java Version (If not docker):

### Example flow

_No response_

### PR DESCRIPTION
**Pull request by Skraye**
close #360
close #1252

### CODE REVIEW

File: core/src/main/java/io/kestra/core/repositories/LogRepositoryInterface.java [hunk 1]
```diff
@@ -5,17 +5,23 @@
...
+
     List<LogEntry> findByExecutionIdAndTaskId(String executionId, String taskId, Level minLevel);
 
+    ArrayListTotal<LogEntry> findByExecutionIdAndTaskId(String executionId, String taskId, Level minLevel, Pageable pageable);
+
     List<LogEntry> findByExecutionIdAndTaskRunId(String executionId, String taskRunId, Level minLevel);
 
+    ArrayListTotal<LogEntry> findByExecutionIdAndTaskRunId(String executionId, String taskRunId, Level minLevel, Pageable pageable);
+
     ArrayListTotal<LogEntry> find(
```
Comment (loicmathieu) [inline core/src/main/java/io/kestra/core/repositories/LogRepositoryInterface.java:25]: Are the non-pageable methods still used ?

File: webserver/src/main/java/io/kestra/webserver/controllers/LogController.java [hunk 2]
```diff
@@ -62,25 +66,49 @@ public PagedResults<LogEntry> find(
 
     @ExecuteOn(TaskExecutors.IO)
     @Get(uri = "logs/{executionId}", produces = MediaType.TEXT_JSON)
-    @Operation(tags = {"Logs"}, summary = "Get logs for a specific execution")
-    public List<LogEntry> findByExecution(
+    @Operation(tags = {"Logs"}, summary = "Get logs for a specific execution, taskrun or task")
+    public PagedResults<LogEntry> findByExecution(
+        @Parameter(description = "The current page") @QueryValue(defaultValue = "1") int page,
+        @Parameter(description = "The current page size") @QueryValue(defaultValue = "50") int size,
+        @Parameter(description = "The sort of current page") @Nullable @QueryValue List<String> sort,
```
Comment (loicmathieu) [inline webserver/src/main/java/io/kestra/webserver/controllers/LogController.java:73]: Usually sort is optional with a sane default.

**Review by loicmathieu — CHANGES_REQUESTED** [review]

File: core/src/main/java/io/kestra/core/repositories/LogRepositoryInterface.java (hunk 1 above)
Comment (Skraye) [reply core/src/main/java/io/kestra/core/repositories/LogRepositoryInterface.java:25]: They are used for the download methods

**Review by loicmathieu — APPROVED** [review]
Backend changes look good to me.
I didn't check the frontend part so a review or this part must be done before merging.

**Review by tchiotludo — CHANGES_REQUESTED** [review]
The experience works well on a finished execution, even if I'm wondering about log order (last to first or first to last). 

But In case of a running execution, it commpletely freeze the browser as soon as I want to see logs: 

image


with this flow 

```yaml 
id: largeLogs
namespace: com.kestra.lde

tasks:
  - id: parrallel
    type: io.kestra.core.tasks.flows.Parallel
    tasks:
      - id: first
        type: io.kestra.core.tasks.scripts.Bash
        commands:
          - "for i in `seq 0 1000`; do echo $i; done"
      - id: second
        type: io.kestra.core.tasks.scripts.Bash
        commands:
          - "for i in `seq 0 1000`; do echo $i; done"
```

**Pull request merged**